src/scripts/bttvIntegration.js

```diff
@@ -0,0 +1,137 @@
+        if (!input || typeof input !== "string"){ areTruthyStrings = false }
+    }
+    return areTruthyStrings
+}
+
+
+/**
+ * @param {string} username twitch username, case insensitive
+ * @returns {string} twitch user id
+ */
+async function getTwitchUserId(username, fetchFunction){ 
+    if (!username || typeof username !== "string"){ 
+        return null;
+    }
+    fetchFunction = fetchFunction || defaultFetch
+    const apiBaseUrl = "https://streamraiders.tips/_functions/getTwitchProfileData/";
+    const urlToQuery = apiBaseUrl + username;
+    const twitchUserData = await fetchFunction(urlToQuery);
+    return new Promise( resolve => { 
+        resolve(twitchUserData.id)
+    })
+}
+
+
+/**
+ * @param {string} userId twitch's user ID 
+ * @param {Function} fetchFunction OPTIONAL - a specific fetch function that returns a promise
+ * that resolves to a json of the fetch data.
+ * @returns {Object} key is the friendly emote name, value is the BTTV emote id
+ */
+async function getBttvChannelEmoteDict(userId, fetchFunction){
+    if ( !userId || typeof userId !== "string") {return null} 
+    fetchFunction = fetchFunction || defaultFetch
+    const apiBaseUrl = "https://api.betterttv.net/3/cached/users/twitch/"
+    const urlToQuery = apiBaseUrl + userId;
+    const bttvChannelData = await fetchFunction(urlToQuery);
+    const emoteDict = convertBttvChannelDataToEmoteDict(bttvChannelData);
+    return new Promise( resolve=> { 
+        resolve(emoteDict);
+    })
+}
+
+
+/**
+ * @param {string} url url to retrieve json from
+ * @returns {Object} json data received from fetch
+ */
+async function defaultFetch(url){ 
+    const response = await fetch(url);
+    const json = await response.json();
+    return new Promise(resolve => { 
+        resolve(json);
+    })
+}
+
+
+/**
+ * @param {Object} channelData  data structure provided by BTTV's api for channel queries
+ * @returns {Object} key is the friendly emote name, value is the BTTV emote id
+ */
+function convertBttvChannelDataToEmoteDict(channelData){ 
+    if (!channelData || typeof channelData !== "object") { return null };
+    let emotesArray = []
+    if (channelData.channelEmotes) { emotesArray.push(...channelData.channelEmotes) }
+    if (channelData.sharedEmotes) { emotesArray.push(...channelData.sharedEmotes)}
+    let emotesDict = {};
+    for (emote of emotesArray){ 
+        const emoteName = emote.code;
+        emotesDict[emoteName] = emote.id;
+    }
+    return emotesDict
+}
+    
+
+/**
+ * @param {string} bttvEmoteId  BTTV's uid for the emote
+ * @returns {string} url for the emote's image source
+ */
+function getBttvImageUrl(bttvEmoteId){ 
+    if (!bttvEmoteId || typeof bttvEmoteId !== "string"){ return null}
+    return url = `https://cdn.betterttv.net/emote/${bttvEmoteId}/3x`
+}
+
+
+
+/**
+ * @param {Object} dictObject  key is friendly emote name, value is the BTTV emote id
+ * @returns {Object} same as input - key is friendly name, value is id
```

Comment:
We can get a touch more specific again with `Object<string, string>`

Reply:
Perfect! I saw the other mention of it too, I'll be clearer about those.  Haven't worked as much with strongly typed languages so it wasn't immediately obvious before, but it makes perfect sense. Will do going forward!